wpilibj/src/main/java/edu/wpi/first/wpilibj/DigitalOutput.java

```diff
@@ -80,12 +80,15 @@ public int getChannel() {
   }
 
   /**
-   * Generate a single pulse. There can only be a single pulse going at any time.
+   * Output a single pulse on the digital output line.
    *
-   * @param pulseLength The length of the pulse.
+   * <p>Send a single pulse on the digital output line where the pulse duration is specified in
+   * microseconds. Maximum of 65535 microseconds
+   *
+   * @param pulseLengthMicroseconds The pulse length in microseconds
    */
-  public void pulse(final double pulseLength) {
-    DIOJNI.pulse(m_handle, pulseLength);
+  public void pulse(final int pulseLengthMicroseconds) {
+    DIOJNI.pulse(m_handle, pulseLengthMicroseconds);
```

Comment:
Is there a reason to change this from natural units (seconds) to integer microseconds?  The *1000 conversion is exact and we generally prefer natural units, particularly on user-facing APIs.

Reply:
I can change it back. Just seemed a bit better, because it was so limited.

Reply:
Yeah, let's change it back (I would recommend changing it back at the HAL layer too).  We use double for other things that are precision limited, e.g. PWM duty cycle.

Reply:
Fixed